syntax = "proto3";

package video;

service VideoService {
    rpc UploadVideo (stream VideoChunk) returns (UploadResponse);
    rpc GetVideo (VideoRequest) returns (stream VideoChunk);
}

message VideoChunk {
    bytes content = 1;
    string video_id = 2;
}

message VideoRequest {
    string video_id = 1;
}

message UploadResponse {
    string video_id = 1;
    bool success = 2;
    string message = 3;
}